plugins/modules/ec2_ami_info.py

```diff
@@ -254,38 +249,29 @@ def build_request_args(executable_users, filters, image_ids, owners):
 
 def get_images(ec2_client, request_args):
     try:
-        images = ec2_client.describe_images(aws_retry=True, **request_args)
-    except (ClientError, BotoCoreError) as err:
-        raise AmiInfoFailure(err, "error describing images")
+        images = describe_images(ec2_client, **request_args)
+    except AnsibleEC2Error as e:
+        raise AmiInfoFailure(e, "error describing images")
     return images
 
 
-def get_image_attribute(ec2_client, image_id):
-    try:
-        launch_permissions = ec2_client.describe_image_attribute(
-            aws_retry=True, Attribute="launchPermission", ImageId=image_id
-        )
-    except (ClientError, BotoCoreError) as err:
-        raise AmiInfoFailure(err, "error describing image attribute")
-    return launch_permissions
-
-
 def list_ec2_images(ec2_client, module, request_args):
-    images = get_images(ec2_client, request_args)["Images"]
+    images = get_images(ec2_client, request_args)
     images = [camel_dict_to_snake_dict(image) for image in images]
 
     for image in images:
         try:
-            image_id = image["image_id"]
             image["tags"] = boto3_tag_list_to_ansible_dict(image.get("tags", []))
             if module.params.get("describe_image_attributes"):
-                launch_permissions = get_image_attribute(ec2_client, image_id).get("LaunchPermissions", [])
+                launch_permissions = describe_image_attribute(
+                    ec2_client, attribute="launchPermission", image_id=image["image_id"]
+                ).get("LaunchPermissions", [])
                 image["launch_permissions"] = [camel_dict_to_snake_dict(perm) for perm in launch_permissions]
-        except is_boto3_error_code("AuthFailure"):
-            # describing launch permissions of images owned by others is not permitted, but shouldn't cause failures
-            pass
-        except (ClientError, BotoCoreError) as err:  # pylint: disable=duplicate-except
-            raise AmiInfoFailure(err, "Failed to describe AMI")
+        except AnsibleEC2Error as e:
+            if e.is_boto3_error_code("AuthFailure"):
+                # describing launch permissions of images owned by others is not permitted, but shouldn't cause failures
+                pass
+            raise AmiInfoFailure(e, "Failed to describe AMI")
```

Comment:
This doesn't do what you think it does.  Pass is just a "do nothing", the pass will do nothing, you'll leave the if block and the raise is still going to be triggered.

Either reverse the logic
```
except AnsibleEC2Error as e:
    if not e.is_boto3_error_code("AuthFailure"):
        raise AmiInfoFailure(e, "Failed to describe AMI")
```

Or handle this logic in module_utils.ec2 with something like:
```
class AnsibleEC2PermissionsError(AnsibleEC2Error):
    pass

@EC2ImageErrorHandler.list_error_handler("describe image attribute", {})
@AWSRetry.jittered_backoff()
def describe_image_attribute(client, image_id: str, attribute: str) -> Optional[Dict[str, Any]]:
    try:
        return client.describe_image_attribute(Attribute=attribute, ImageId=image_id)
    except is_boto3_error_code("AuthFailure") as e:
        raise AnsibleEC2PermissionsError(message="Failed to describe Image Attributes", exception=e) from e
```

Which means you can simplify the logic here to:
```suggestion
        except AnsibleEC2PermissionsError:
            pass
```